src/__tests__/rust-plugin-test.js

```diff
@@ -0,0 +1,40 @@
+import * as shelljs from 'shelljs';
+import path from 'path';
+
+const whichCargo = shelljs.which('cargo');
+const rustInstalled = whichCargo && whichCargo.code === 0;
+
+const rustCalcDir = path.join(__dirname, 'rust-calc');
+
+if (rustInstalled) {
+  describe('rust plugin', () => {
+    beforeAll(() => {
+      shelljs.exec('make', {
+        cwd: path.join(rustCalcDir, 'calc-syntax'),
+      });
+    }, 10000);
+
+    it('calc rust example should build, also output must match expected value', () => {
```

Comment:
So this fails now out of the box.

```
 FAIL  src/__tests__/rust-plugin-test.js
  ● rust plugin › calc rust example should build, also output must match expected value

    expect(received).toEqual(expected)

    Expected value to equal:
      0
    Received:
      101

      at Object.<anonymous> (src/__tests__/rust-plugin-test.js:24:30)
          at new Promise (<anonymous>)
      at Promise.resolve.then.el (node_modules/p-map/index.js:42:16)
      at process.internalTickCallback (internal/process/next_tick.js:77:7)
```

What should it be installed for this?

Reply:
Also getting this when trying to run `cargo run`:

```
.dmitrys:calc-bin$ cargo run
warning: unused manifest key: package.edition
warning: unused manifest key: package.edition
   Compiling calc-syntax v0.1.0 (file:///Users/dmitrys/syntax/src/__tests__/rust-calc/calc-syntax)
   Compiling onig_sys v69.1.0
error[E0658]: use of unstable library feature 'fs_read_write' (see issue #46588)
  --> /Users/dmitrys/.cargo/registry/src/github.com-1ecc6299db9ec823/onig_sys-69.1.0/build.rs:95:9
   |
95 |         fs::write(config_h, format!("
   |         ^^^^^^^^^

error: aborting due to previous error

error: Could not compile `onig_sys`.
```

Since this test relies on `cargo` to be installed we should probably check in the test if it's available. Otherwise the `npm test` currently fails when check out the repo.

Reply:
Looks like minimum compatible rust version for onig is 1.26.0 (latest is 1.35.0).
I'll add additional check for this

Reply:
Also maybe helpful: command for update rust toolchain: `rustup update`

Reply:
Made PR to address this issue in https://github.com/DmitrySoshnikov/syntax/pull/83